### ISSUE TITLE
Need for new scattering models

### ISSUE THREAD

**Issue by KAHR-Alpha** [KAHR-Alpha/Aether#10]
The implementation of the Lambertian, Guassian and ABg scattering models has been requested. ( see https://support.zemax.com/hc/en-us/articles/1500005575662-What-scattering-models-are-available-in-OpticStudio- )

**Comment by KAHR-Alpha** [comment]
Starting documentation:

* [https://en.wikipedia.org/wiki/Bidirectional_reflectance_distribution_function](Bidirectional reflectance distribution function)
* [Theory for off-specular reflection from roughened surfaces](http://www-inst.eecs.berkeley.edu/~cs294-13/fa09/lectures/TorranceSparrow.pdf)
* [A reflectance model for computer graphics](https://dl.acm.org/doi/pdf/10.1145/357290.357293)
* [Approximated scatter models for stray light analysis](https://photonengr.com/wp-content/uploads/2014/10/ApproximatedScatter1.pdf)

**Comment by KAHR-Alpha** [comment]
@iromero91 it seems everywhere I look the models are reflective only. Did you use them in transmission?

**Comment by iromero91** [comment]
The main purpose i need them for is total internal reflection, but transmission too. The models in that zemax doc just affect the ray as it would come out of the surface, either refracted or reflected, shouldn't matter.

**Comment by KAHR-Alpha** [comment]
Yeah eventually that's what I saw on the page about fitting the ABg coefficients. One thing I still don't get is how reflection and transmission are coupled through the model.

Total internal reflection is a good example. One would expect some light to be scattered through the surface, but not according to that algorithm.
